purchases/src/main/kotlin/com/revenuecat/purchases/SyncPurchasesHelper.kt

```diff
@@ -0,0 +1,77 @@
+package com.revenuecat.purchases
+
+import com.revenuecat.purchases.common.BillingAbstract
+import com.revenuecat.purchases.common.LogIntent
+import com.revenuecat.purchases.common.ReceiptInfo
+import com.revenuecat.purchases.common.debugLog
+import com.revenuecat.purchases.common.errorLog
+import com.revenuecat.purchases.common.log
+import com.revenuecat.purchases.identity.IdentityManager
+import com.revenuecat.purchases.models.StoreTransaction
+import com.revenuecat.purchases.strings.PurchaseStrings
+
+internal class SyncPurchasesHelper(
+    private val billing: BillingAbstract,
+    private val identityManager: IdentityManager,
+    private val postReceiptHelper: PostReceiptHelper
+) {
+    fun syncPurchases(
+        isRestore: Boolean,
+        onSuccess: () -> Unit,
+        onError: (PurchasesError) -> Unit
+    ) {
+        log(LogIntent.DEBUG, PurchaseStrings.SYNCING_PURCHASES)
+
+        val appUserID = identityManager.currentAppUserID
+
+        billing.queryAllPurchases(
+            appUserID,
+            onReceivePurchaseHistory = { allPurchases ->
+                if (allPurchases.isNotEmpty()) {
+                    val lastPurchase = allPurchases.last()
+                    val errors: MutableList<PurchasesError> = mutableListOf()
+                    fun handleLastPurchase(currentPurchase: StoreTransaction, lastPurchase: StoreTransaction) {
```

Comment:
I'm usually not a big fan of local functions, but I thought it was useful here. Lmk if you have any concerns though!